### PR TITLE
Set Shovel BuildFireMode to automatic

### PR DESCRIPTION
**Pull request by LordDz**
Simple change, but so nice for all logi players.
Nothing more than that, really.

### CODE REVIEW

**Review by shiftycats — CHANGES_REQUESTED** [review]

File: DH_Equipment/Classes/DHShovelBuildFireMode.uc [hunk 1]
```diff
@@ -3,7 +3,7 @@
 // Darklight Games (c) 2008-2023
 //==============================================================================
 
-class DHShovelBuildFireMode extends DHWeaponFire;
+class DHShovelBuildFireMode extends DHAutomaticFire;
```
[the remark this answers is not in the document]
Comment (LordDz) [reply DH_Equipment/Classes/DHShovelBuildFireMode.uc:6]: I tried doing that, but then it plays the digging animation + sound but it never seems to actually build anything, it just shovels on with no progress done to the construction.

[the remark this answers is not in the document]
Comment (shiftycats) [reply DH_Equipment/Classes/DHShovelBuildFireMode.uc:6]: Right. When the release flag is false, `ModeDoFire` event is continuously called while the mouse button is pressed. And every time it's run, it clears the construction actor. That's why there's no construction progress.

https://github.com/DarklightGames/DarkestHour/blob/2533f7e657a91fb375380531b32c55018abfa6f9/DH_Equipment/Classes/DHShovelBuildFireMode.uc#L62-L64

To fix this, add a `event ModeDoFire();` line in the `Building` state to override the function so it's not doing anything while digging is in progress.

[the remark this answers is not in the document]
Comment (shiftycats) [reply DH_Equipment/Classes/DHShovelBuildFireMode.uc:6]: Also, it's a good idea to add a `StopFire` call to stop `ModeDoFire` from being called needlessly when construction is done or the player is no longer looking at it.

```diff
 event ModeDoFire()
 {
     Construction = none;
 
     if (AllowFire())
     {
         GotoState('Building');
     }
+    else
+    {
+        Weapon.StopFire(ThisModeNum);
+    }
 }
```

[the remark this answers is not in the document]
Comment (LordDz) [reply DH_Equipment/Classes/DHShovelBuildFireMode.uc]: Updated it with your suggestions @shiftycats.

**Review by shiftycats — APPROVED** [review]
Looks good.

**Pull request merged**